tests/test_features/test_pcoords.py

```diff
@@ -17,134 +18,279 @@
+    def test_alpha(self):
+        """
+        Test image closeness on opaque alpha for random 3 class dataset
+        """
+        visualizer = ParallelCoordinates(alpha=1.0)
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
+    def test_alpha_fast(self):
+        """
+        Test image closeness on opaque alpha for random 3 class dataset in fast mode
+        """
+        visualizer = ParallelCoordinates(alpha=1.0, fast=True)
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
+    def test_labels(self):
+        """
+        Test image closeness when class and feature labels are supplied
+        """
+        visualizer = ParallelCoordinates(
+            classes=['a', 'b', 'c'], features=['f1', 'f2', 'f3', 'f4', 'f5']
+        )
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
         visualizer.poof()
         self.assert_images_similar(visualizer)
 
-    def test_normalized_pcoords_invalid_arg(self):
+    def test_labels_fast(self):
+        """
+        Test image closeness when class and feature labels are supplied in fast mode
+        """
+        visualizer = ParallelCoordinates(
+            classes=['a', 'b', 'c'], features=['f1', 'f2', 'f3', 'f4', 'f5'], fast=True
+        )
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer)
+
+    def test_normalized_l2(self):
+        """
+        Test image closeness on l2 normalized 3 class dataset
+        """
+        visualizer = ParallelCoordinates(normalize='l2')
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
+    def test_normalized_l2_fast(self):
+        """
+        Test image closeness on l2 normalized 3 class dataset in fast mode
+        """
+        visualizer = ParallelCoordinates(normalize='l2', fast=True)
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
+    def test_normalized_minmax(self):
+        """
+        Test image closeness on minmax normalized 3 class dataset
+        """
+        visualizer = ParallelCoordinates(normalize='minmax')
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
+    def test_normalized_minmax_fast(self):
+        """
+        Test image closeness on minmax normalized 3 class dataset in fast mode
+        """
+        visualizer = ParallelCoordinates(normalize='minmax', fast=True)
+        visualizer.fit_transform(self.dataset.X, self.dataset.y)
+        visualizer.poof()
+        self.assert_images_similar(visualizer, tol=0.25)
+
```

Comment:
Maybe this is overkill, but is it worth it to test that "fast" is at least as fast as "slow" given the same dataset?

Reply:
This would be tough to do in our current test framework, we'd have to implement some benchmarking tools;  but you're right - probably as a result of this suite of issues it is something we should look into. Have added #460 in response.